hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMStateMachine.java

```diff
@@ -89,4 +99,15 @@ private Message process(final SCMRatisRequest request)
     }
   }
 
+  @Override
+  public long takeSnapshot() throws IOException {
+    LOG.info("Current Snapshot Index {}", getLastAppliedTermIndex());
+    long startTime = Time.monotonicNow();
+    TermIndex lastTermIndex = getLastAppliedTermIndex();
+    long lastAppliedIndex = lastTermIndex.getIndex();
+    transactionBuffer.flush();
+    LOG.debug("SCM takeSnapshot: {} took {} ms",
+        Time.monotonicNow() - startTime);
+    return lastAppliedIndex;
```

Comment:
Maybe flush should return SCMTransactionInfo, and this return SCMTransactionInfo#transactionIndex ?

Reply:
Code changes a bit. Maybe take another look?